app/controllers/api/v1/api_controller.rb

```diff
@@ -4,6 +4,13 @@ class ApiController < ApplicationController
       include DeviseTokenAuth::Concerns::SetUserByToken
 
       before_action :authenticate_user!, unless: :devise_controller?
+      before_action :set_locale
+
+      private
+
+      def set_locale
+        I18n.locale = current_user.try(:locale) || I18n.default_locale
```

Comment:
Since ruby 2.3.0, you can use safe navigation instead of try. Here it would be `current_user&.locale || I18n.default_locale`